### PR TITLE
Add support for zipkin

### PR DESCRIPTION
**Pull request by matt-richardson**
# Background

The plugin currently supports honeycomb and "custom".
But, I'm less and less convinced that the "custom" is usable.

I tried to get it to work with zipkin, but it couldn't get it to work.

Also, I wanted to check out zipkin and see if it's a good tool.

# Results

This PR adds support for Zipkin as a target.

image

image

image

# How to review this PR

Depending on how enthusiastic you want to be, it might be worth pulling it down and testing it.

You can start a zipkin instance by going
```
 docker run -d -p 9411:9411 openzipkin/zipkin
```

and set your settings to:
* service: `zipkin`
* endpoint: `http://localhost:9411/`

(also double check I haven't broken the main honeycomb flow)

# Pre-requisites
- [x] I have considered informing or consulting the right people
- [x] I have considered appropriate testing for my change.

### CODE REVIEW

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/BuildOverviewExtensionController.java [hunk 1]
```diff
@@ -90,29 +90,50 @@ protected ModelAndView doHandle(@NotNull HttpServletRequest request, @NotNull Ht
...
+    @NotNull
+    private ModelAndView getHoneycombModelAndView(SBuild build, Map<String, String> params, String traceId) {
+        final ModelAndView mv = new ModelAndView(pluginDescriptor.getPluginResourcesPath("buildOverviewHoneycombExtension.jsp"));
+
+        var model = mv.getModel();
+        model.put("team", params.get(PROPERTY_KEY_HONEYCOMB_TEAM));
+        model.put("dataset", params.get(PROPERTY_KEY_HONEYCOMB_DATASET));
+        model.put("traceId", traceId);
+
+        //we pad the time to ensure that we get all the spans, just in case we get a slight diff in the
```
Comment (casstait) [inline server/src/main/java/com/octopus/teamcity/opentelemetry/server/BuildOverviewExtensionController.java:129]: Do we need this padding for zipkin too?

Comment (matt-richardson) [reply server/src/main/java/com/octopus/teamcity/opentelemetry/server/BuildOverviewExtensionController.java:129]: Nah, as we can link directly to the trace in zipkin

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/ProjectConfigurationSettingsController.java [hunk 2]
```diff
@@ -133,8 +133,8 @@ public boolean validate(@NotNull ActionErrors errors) {
             if (!this.service.isPresent())
                 errors.addError("service", "Service must be set!");
             else {
-                if (!this.service.get().equals(OTELService.HONEYCOMB) && !this.service.get().equals(OTELService.CUSTOM))
-                    errors.addError("service", "Service must be set to either 'honeycomb.io' or 'custom'!");
+                if (!this.service.get().equals(OTELService.HONEYCOMB) && !this.service.get().equals(OTELService.CUSTOM) && !this.service.get().equals(OTELService.ZIPKIN))
```
Comment (casstait) [inline server/src/main/java/com/octopus/teamcity/opentelemetry/server/ProjectConfigurationSettingsController.java:136]: Just a suggestion, maybe its time to move these many `ifs` to different methods or using switches

Comment (matt-richardson) [reply server/src/main/java/com/octopus/teamcity/opentelemetry/server/ProjectConfigurationSettingsController.java:136]: Yeah, completely agree.
I'm thinking of hanging some kind of `PropertyAdapter`s off the service enum - that would clean this up quite nicely.

Comment (matt-richardson) [reply server/src/main/java/com/octopus/teamcity/opentelemetry/server/ProjectConfigurationSettingsController.java:136]: But... later... Not keen to do it as part of this PR.

File: server/src/main/resources/buildServerResources/buildOverviewZipkinExtension.jsp [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
...XECBEMFeQrMHqJRKkS3U8QDHM0WAVGo+Z0Vkj0gvxAgKECJIEydrkXytBuk7H3V3UYHVrMgMdxMzoznJ4AQFCBHUIkrVQ+bUFDVahA5gZ3ZmKazSn6PYBCBG0W5TucjRYhfp4THGG4peY0QFCBI0UJBqsQhr3K0YqbsOMDhAiaKYg2d1uLwnSCBfZmH9fsSCZKTW/V5ypufE7UgHtvraQAmiHINFgFW5TbKt5sAMiBAgR+BQkGqyWjxsV39S47664l3QAQgShCNLrihP0x9UVox0bRIuGvfO5SrGxxrmn4iFSAo2Ed0TQSEGyti3DZw/uMUnHfgoTJxqs5hdbdm0bncdpbJ8jHYAQQZ4EyZpXjo0brP5IcaJipYC/8qeM2jyYGd2Figkay5dJByBEkGdBsndGbRusmiCtEeBXpedZhC06scUnkzV2fyUdgBBBkQSpbYPV3oqTXHkbrIaI3TBMU5yNGR0gRFB0QbJeYxdLkH6u44Eu6tZAg1V/2KIS8wI6FzM6QIigbIJkrV+ukSBdq+MeLmqwuhWZyYy3XbTcfgZmdIAQQdkFyZYF32IRN1g9TbETmWka5gVkKxrPw4wOECKA+UXpLvffBqvmibQ3WWkYtvBgguKnmNEBQgSQLkjWYHUfCdJmLlrUQIPV9mNeQOaGejFmdBD0eU8KIFBBelTRS3/c0C6kLtpcCbVhXkCHK9ZXDs9HhAAhAuiYID0XN1hdy0V7XNh8msyc7uiK7srZz3FEBYQIoLGC9Je4waptiG1xNFhti3kB9VRsrhxdhSEd5A3eEUHeBMk2XB4/e3CPMToepzhGsWRJ0/EHxVjl5FZmBiBEANkLUiMbrOZtKfNvYgG6i5kAhTifSQHkXJDeVYzVH1dTDFa0p0VNXt6l3KTood+7GyIECBFAeIL0oWKqizyRrEJ6pSA/zTb9WgcKe/+zT7y8HaBQ8GgOiiZIRWmwasvVr1SM0W96hpEFhAggf4LUtsHqQYpTXD4arJoXkH1newf0IiMJCBFA/gXJljJfLUG6Rsc9XdTPLsQGq+YF9DMXuaG+ysgBQgRQPEGydy03W0iUdowFaccAvprthzrfRW6oeAEBQgRQElG6U4c74warwxXdPHwN8wKapZiEFxAAQMmRIK2a8ed1VSxF5gEAAAAC4P8FGABDGMmeCJOEkgAAAABJRU5ErkJggg=="/>
```
Comment (casstait) [inline server/src/main/resources/buildServerResources/buildOverviewZipkinExtension.jsp:8]: 😂 gross

Comment (matt-richardson) [reply server/src/main/resources/buildServerResources/buildOverviewZipkinExtension.jsp:8]: an SVG with an embedded image? _I think they're doing it wrong 🤦 _

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/helpers/OTELHelperImpl.java [hunk 4]
```diff
@@ -31,9 +25,7 @@ public class OTELHelperImpl implements OTELHelper {
     private final Tracer tracer;
     private final ConcurrentHashMap<String, Span> spanMap;
 
-    public OTELHelperImpl(Map<String, String> headers, String exporterEndpoint) {
-        var spanProcessor = buildSpanProcessor(headers, exporterEndpoint);
-
+    public OTELHelperImpl(SpanProcessor spanProcessor) {
```
Comment (casstait) [inline server/src/main/java/com/octopus/teamcity/opentelemetry/server/helpers/OTELHelperImpl.java:28]: Love this change 👍

**Review by casstait — APPROVED** [review]
Test Locally and works great. Honeycomb still works too. 
Good 2 Go! 👍 

Excited for Jaeger and Sumo next :trollface:

**Pull request merged**